src/main/java/org/testng/xml/XmlSuite.java

```diff
@@ -333,7 +334,9 @@ public String getName() {
    * @param name The name to set.
    */
   public void setName(String name) {
-    m_name = name;
+    if (Strings.isNotNullAndNotEmpty(name)) {
+      m_name = name;
+    }
```

Comment:
what else? any warning or exception should be there?

Reply:
Good catch, I agree. An exception could be better than an unexpected behavior.

Reply:
@juherr - added a exception Can we get this merged ?